_upgrading/sopel-8.md

```diff
@@ -0,0 +1,420 @@
+maximum Sopel versions in their requirements. If all plugins a bot owner wishes
+to install contain this metadata, they can ask `pip` to install Sopel and all of
+its plugins in a single command, and `pip`'s dependency solver will figure out
+which version of Sopel satisfies all of the plugins' needs.
+
+We do everything we can to keep breaking API changes to *only* major versions,
+so a version range like `sopel>=7.1,<9` is typically safe—at least the earliest
+Sopel version your plugin supports, up to (but excluding) the next unreleased
+major version.
+
+In the vast majority of cases, removed API features will first go through a
+deprecation period, during which Sopel will log warnings whenever the deprecated
+functionality is used by a plugin. Rarely, we might need to remove an API
+feature without going through this process—but that's a last-resort option.
+
+### Command names are now literal
+
+To prevent conflicts with the default capture groups Sopel defines for commands
+and prepare for future enhancements to plugin/command management, characters
+with special meaning in regular expressions are now escaped in command names.
+
+Documentation for `@sopel.plugin.command()` suggests alternatives based on the
+use case you need to adapt for Sopel 8.
+
+### Enums everywhere
+
+_(Insert that classic Buzz Lightyear meme here, if you want.)_
+
+Modernizing Sopel to run only on currently-supported Python versions let us do
+quite a bit of work under the hood, but we were also able to start improving how
+some parts of the API work. Using `Enum`s where it makes sense is part of that.
+
+The following parts of Sopel's API are now expressed as some type of `Enum`:
+
+* `sopel.formatting.colors`
+* `sopel.tools.events`
+
+### Identifier casemapping
+
+Any reasonably modern IRC server advertises its method for normalizing nicknames
+to every client that connects. Sopel 8 builds on the `bot.isupport` mechanism
+added in Sopel 7 and makes use of the advertised `CASEMAPPING` value when
+comparing and normalizing nicknames itself.
+
+To support handling this under the hood, the bot's `config.core.nick` setting is
+now stored and returned as a normal `str`. It and any other strings that you'd
+like to treat as nicknames should be passed through the `bot.make_identifier()`
+method to get the same `Identifier` type you're used to from Sopel 7, configured
+to use the IRC server's advertised casemapping method.
+
+### Time-handling changes
+
+`trigger.time` is now an "aware" `datetime` object, meaning it has a UTC offset
+associated with it. Comparison or arithmetic operations between "aware" and
+"naive" `datetime` objects are not allowed; code that manipulates `trigger.time`
+values will need to be updated for Sopel 8.
+
+The fallback format string used by `sopel.tools.time.format_time()` if no other
+source provides one changed from outputting the timezone name (`UTC`) to the UTC
+offset (`+0000`).
```

Comment:
Nit-pick: this sentence is a little hard to read even as a native speaker, feels like it could use a rewrite.

Reply:
I can see it… Woof, past-me. Retry, longer but hopefully clearer:

```suggestion
The fallback format string used by `sopel.tools.time.format_time()` changed from
outputting the timezone name (`UTC`) to the UTC offset (`+0000`). This format
string is used only if there is no preferred format set for the nick, channel,
_and/or_ config passed to the function.
```

Reply:
Much more legible!